File: 2023/go/aoc_2023/day1/day1_test.go
package day1

import (
	"os"
	"testing"
)

func TestPart1(t *testing.T) {
	input, err := os.ReadFile("input.txt")
	if err != nil {
		t.Fatal(err)
	}
	res, err := Part1(string(input))
	if err != nil {
		t.Fatal(err)
	}
	if res != 56049 {
		t.Logf("res: %d", res)
		t.Fatalf("Expected 56049, got %d", res)
	}
}

func TestPart2(t *testing.T) {
	input, err := os.ReadFile("input.txt")
	if err != nil {
		t.Fatal(err)
	}
	res, err := Part2(string(input))
	if err != nil {
		t.Fatal(err)
	}
	if res != 54530 {
		t.Logf("res: %d", res)
		t.Fatalf("Expected 54530, got %d", res)
	}
}

func TestEx1(t *testing.T) {
	input, err := os.ReadFile("ex1.txt")
	if err != nil {
		t.Fatal(err)
	}
	res, err := Part1(string(input))
	if err != nil {
		t.Fatal(err)
	}
	if res != 142 {
		t.Fatalf("Expected 142, got %d", res)
	}
}

func TestEx2(t *testing.T) {
	input, err := os.ReadFile("ex2.txt")
	if err != nil {
		t.Fatal(err)
	}
	res, err := Part2(string(input))
	if err != nil {
		t.Fatal(err)
	}
	if res != 281 {
		t.Fatalf("Expected 3, got %d", res)
	}
}

func TestGetNums(t *testing.T) {
	input, err := os.ReadFile("ex2.txt")
	if err != nil {
		t.Fatal(err)
	}
	nums, err := GetNums(string(input))
	if err != nil {
		t.Fatal(err)
	}
	for num := range nums {
		t.Logf("num: %v", num)
	}
}

File: 2023/1/day1_py/src/py_day1/day1_test.py
from . import day1


def test_part1():
    with open("1/input.txt") as f:
        inp = f.read()
    ans = day1.part1(inp)
    assert ans == 56049


def test_part2():
    with open("1/input.txt") as f:
        inp = f.read()
    ans = day1.part2(inp)
    assert ans == 54530


def test_ex1():
    with open("1/ex1.txt") as f:
        inp = f.read()
    ans = day1.part1(inp)
    assert ans == 142


def test_ex2():
    with open("1/ex2.txt") as f:
        inp = f.read()
    ans = day1.part2(inp)
    assert ans == 281
